sklearn/_isotonic.pyx

```diff
@@ -90,13 +90,12 @@ def _make_unique(np.ndarray[dtype=floating] X,
     cdef floating current_weight = 0
     cdef floating y_old = 0
     cdef int i = 0
-    cdef int current_count = 0
     cdef int j
     cdef floating x
     cdef int n_samples = len(X)
     for j in range(n_samples):
         x = X[j]
-        if x != current_x:
+        if x - current_x >= 1e-15:
```

Comment:
I think it would be cleaner/safer to use an eps that depends on the precision level of X.

```python
cdef floating eps = np.ffinfo(X.dtype).eps

for j in range(n_samples):
    x = X[j]
    if x - current_x >= eps:
....
```

Reply:
Actually maybe we need to use tiny `instead` of `eps`, see: https://numpy.org/doc/stable/reference/generated/numpy.finfo.html

IEEE floating point values (float32, float64) are denser around 0 than around 1.